subprojects/docs/src/samples/build-organization/standalone-precompiled-plugins/kotlin/src/main/kotlin/com.example.service.gradle.kts

```
@@ -0,0 +1,30 @@
+// tag::plugins[]
+plugins {
+    id("com.example.java-convention")
+}
+// end::plugins[]
+
+val integrationTest by the<SourceSetContainer>().creating {
+    compileClasspath += sourceSets["main"].output + sourceSets["test"].output
+    runtimeClasspath += sourceSets["main"].output + sourceSets["test"].output
+}
+
+val integrationTestTask = tasks.register<Test>("integrationTest") {
+    shouldRunAfter(tasks.named("test"))
+
+    testClassesDirs = sourceSets["integrationTest"].output.classesDirs
+    classpath = sourceSets["integrationTest"].runtimeClasspath
+}
+
+configurations["integrationTestImplementation"].extendsFrom(configurations["testImplementation"])
+configurations["integrationTestRuntimeOnly"].extendsFrom(configurations["testRuntimeOnly"])
+
+// tag::use-java-class[]
+val readmeCheck = tasks.register<com.example.ReadmeVerificationTask>("readmeCheck") {
```

💄 can be simplified by not repeating the task name
```suggestion
val readmeCheck by tasks.registering(com.example.ReadmeVerificationTask::class) {
```